### DISCUSSION TITLE
Quando usar o After e Before?

### DISCUSSION THREAD

**Discussion by Marcelosilva10** [frontendbr/forum#169] [Perguntas e Respostas]
Pessoal tenho essa seguinte dúvida, quando a ser usado After e Before e qual momento é necessário a ser usado?

**Comment by felipefialho** [comment]
Particularmente eu considero o `:after` e o `:before` uma das coisas mais incriveis do CSS, pois te dão um mundo de possibilidades.

Você pode usar para resolver basicamente qualquer questão nos detalhes do layout e até mesmo para usar os atributos do HTML para fazer um [tooltip foda usando apenas CSS](http://www.felipefialho.com/css-components/#component-tooltip). 

E mais uma coisa, lembre-se que com esses pseudo-elementos você consegue transformar um elemento em três. Olha essa minha pen bem antiga (2013, por ai), onde estava justamente testando as possibilidades -> http://codepen.io/LFeh/pen/GxmKH

**Comment by felipecotto** [comment]
Utilizo bastante para detalhes de layout, bordas, linhas, transições css. macaras em imagens, formas.
Realmente é uma das coisas mais incríveis do CSS.

**Comment by felipebernardes** [comment]
@Marcelosilva10 vale ressaltarmos a **diferença** entre usar after e before:
- se você definir `content: ''`, não há diferença alguma.
- se você definir `content: 'foo';` em `element:before`, foo aparecerá **_antes_** de todo o conteúdo do elemento. se definir em `.element:after`, foo aparecerá _**após**_ todo o conteúdo do elemento.

**Comment by leandroruel** [comment]
já utilizei bastante, principalmente para criar shapes apenas com css.

**Comment by kandebonfim** [comment]
@Marcelosilva10 usei recentemente pra criar essas janelinhas:

![captura de tela 2016-06-22 as 14 02 29](https://cloud.githubusercontent.com/assets/8906380/16275863/5092ec92-3882-11e6-8972-acd32c347fb6.png)

A barra com o nome da janela e os 3 botões foram feitos com pseudoelementos. No final o HTML só precisa ter uma div. Eu me amarro quando dá pra fazer esse tipo de coisa.

Link do projeto em que eu usei isso: https://kandebonfim.github.io/show-your-terms/
